core/conn/odbc/src/odbc/nsksrvrcore/srvrothers.cpp

```diff
@@ -4503,78 +4504,107 @@ odbc_SQLSvc_GetSQLCatalogs_sme_(
-                               "('INTERVAL', 103, 0, '{INTERVAL ''', ''' DAY}', NULL, 1, 0, 2, 0, 0, NULL, 'INTERVAL', 0, 0, 'INTERVAL', NULL, 0, 34, 100, 3, 2, 3, 3, 3, 0),"
-                               "('NUMERIC', 2, 128, NULL, NULL, 'precision,scale', 1, 0, 2, 0, 0, 0, 'NUMERIC', 0, 128, 'NUMERIC', 10, -2, -3, 2, NULL, NULL, 0, 0, 3, 0),"
-                               "('NUMERIC UNSIGNED', 2, 128, NULL, NULL, 'precision,scale', 1, 0, 2, 1, 0, 0, 'NUMERIC', 0, 128, 'UNSIGNED NUMERIC', 10, -2, -3, 2, NULL, NULL, 0, 0, 3, 0),"
-                               "('REAL', 7, 7, NULL, NULL, NULL, 1, 0, 2, 0, 0, 0, 'REAL', NULL, NULL, 'REAL', 2, 22, -1, 7, NULL, NULL, 0, 0, 3, 0),"
-                               "('SMALLINT', 5, 5, NULL, NULL, NULL, 1, 0, 2, 0, 0, 0, 'SMALLINT', NULL, NULL, 'SMALLINT', 10, 5, -1, 5, NULL, NULL, 0, 0, 3, 0),"
-                               "('SMALLINT UNSIGNED', 5, 5, NULL, NULL, NULL, 1, 0, 2, 1, 0, 0, 'SMALLINT', NULL, NULL, 'UNSIGNED SMALLINT', 10, 5, -1, -502, NULL, NULL, 0, 0, 3, 0),"
-                               "('TIME', 10, 8, '{t ''', '''}', NULL, 1, 0, 2, NULL, 0, NULL, 'TIME', NULL, NULL, 'TIME', NULL, 8, 6, 9, 2, NULL, 4, 6, 3, 0),"
-                               "('TIMESTAMP', 11, 26, '{ts ''', '''}', NULL, 1, 0, 2, NULL, 0, NULL, 'TIMESTAMP', 0, 6, 'TIMESTAMP', NULL, 19, 16, 9, 3, NULL, 1, 6, 3, 0),"
-                               "('VARCHAR', 12, 32000, '''', '''', 'max length', 1, 1, 3, NULL, 0, NULL, 'VARCHAR', NULL, NULL, 'VARCHAR', NULL, -1, -1, 12, NULL, NULL, 0, 0, 3, 0)"
-                               " ) "
-                               " dt(\"TYPE_NAME\", \"DATA_TYPE\", \"PREC\", \"LITERAL_PREFIX\", \"LITERAL_SUFFIX\", \"CREATE_PARAMS\", \"IS_NULLABLE\", \"CASE_SENSITIVE\", \"SEARCHABLE\","
-                               "\"UNSIGNED_ATTRIBUTE\", \"FIXED_PREC_SCALE\", \"AUTO_UNIQUE_VALUE\", \"LOCAL_TYPE_NAME\", \"MINIMUM_SCALE\", \"MAXIMUM_SCALE\", \"SQL_TYPE_NAME\","
-                               "\"NUM_PREC_RADIX\", \"USEPRECISION\", \"USELENGTH\", \"SQL_DATA_TYPE\", \"SQL_DATETIME_SUB\", \"INTERVAL_PRECISION\", \"DATETIMESTARTFIELD\","
+                case SQL_API_SQLGETTYPEINFO_JDBC :
+                     {
+                         SQLSMALLINT SqlTypeCode_Date;
+                         SQLSMALLINT SqlTypeCode_Time;
+                         SQLSMALLINT SqlTypeCode_TimeStamp;
+                         char condExpression[20] = {0};
+                        
+                         if(APIType == SQL_API_SQLGETTYPEINFO_JDBC)
+                         {
+                             SqlTypeCode_Date = 91;
+                             SqlTypeCode_Time = 92;
+                             SqlTypeCode_TimeStamp = 93;
+                         }
+                         else
+                         {
+                             SqlTypeCode_Date = SQL_DATE;
```

Comment:
@selvaganesang You are right about this. But our driver driver will convert old definition to version 3 value. Basically it is compatible with ODBC 3.0 and 2.1, but relying on how the environment attribute SQL_ATTR_ODBC_VERSION is set by application. This problem was detected by Coast test cases, and Coast expects old value. So I did not realize if there would be any impact by returning the old value. We don't have plenty test cases now to test all considerable conditions, however the coast just looks fine now.
I don't know is there any application is running on our ODBC, which relies on this catalog API now, and it might be running into this compatibility problem. If the SQL_ATTR_ODBC_VERSION is not set properly, it might. If there's any app has the problem, I'd like to know then I will see what I can do about it.
![5 pic](https://cloud.githubusercontent.com/assets/7919914/16660085/87fb2798-449f-11e6-8d22-85c93b77ce5f.jpg)
![6 pic](https://cloud.githubusercontent.com/assets/7919914/16660240/193fcefc-44a0-11e6-808d-e69fb6c3c9b7.jpg)